### ISSUE TITLE
AST differs with multiline splice

### ISSUE THREAD

**Issue by waj** [tweag/ormolu#707]
**Describe the bug**
A multiline splice in a `do` block followed with an argument fails with a different AST

**To Reproduce**
Try to format this code:

```haskell
foo = do
  $( bar
   ) baz
```

**Environment**
 - OS name + version: macOS 10.15.7
 - Version of the code: 
 ```
% ormolu -v
ormolu 0.1.4.1 UNKNOWN UNKNOWN
using ghc-lib-parser 8.10.4.20210206
```

**Comment by amesgen** [comment]
I pushed a fix for this specific problem in [`amesgen/issue-707`](https://github.com/tweag/ormolu/tree/amesgen/issue-707), but the problem goes deeper:

---

Consider this:

<table>
<tr>
<td>

```haskell
foo = 
  case 1 of
    1 -> (+1)
   1
```

</td>
<td>

```haskell
foo = do
  case 1 of
    1 -> (+1)
   1
```

</td>
</tr>
</table>

Note that both are valid Haskell code with `BlockArguments` (`foo == 2`). Formatting them (the second snippet needs `--unsafe`) results in

<table>
<tr>
<td>

```haskell
foo = 
  case 1 of
    1 -> (+1)
  1
```

</td>
<td>

```haskell
foo = do
  case 1 of
    1 -> (+1)
  1
```

</td>
</tr>
</table>

and one can see the problem here.

---

This seems to be a bit similar to #488, and indeed, we could fix this by formatting `case` blocks with four spaces of indentation and its arguments with two spaces:

<table>
<tr>
<td>

```haskell
foo = 
  case 1 of
      1 -> (+1)
    1
```

</td>
<td>

```haskell
foo = do
  case 1 of
      1 -> (+1)
    1
```

</td>
</tr>
</table>

But this would cause an enormous amount of reformatting, so I am not sure if this is a good idea.

**Comment by amesgen** [comment]
Another instance of the same problem:

```haskell
foo = do 
  do
    (+1)
   1
```

**Comment by amesgen** [comment]
The least bad solution for the examples above seems to be indenting the arguments by 1, so e.g.

```haskell
foo = do 
  do
    (+1)
   1
```
is a fixed point. I implemented this in #731, let's see how it goes.
